K2Bridge.Tests.UnitTests/Visitors/RangeAggregationVisitorTests.cs

```diff
@@ -31,8 +30,30 @@ public string RangeVisit_WithAggregation_ReturnsValidResponse(double? from1, dou
                 },
             };
 
-            var visitor = new ElasticSearchDSLVisitor(SchemaRetrieverMock.CreateMockNumericSchemaRetriever());
-            VisitorTestsUtils.VisitRootDsl(visitor);
+            var visitor = VisitorTestsUtils.CreateAndVisitRootVisitor("dayOfWeek", "double");
+            visitor.Visit(rangeAggregation);
+
+            return rangeAggregation.KustoQL;
+        }
+
+        [TestCase(0, 800, 800, 2000, ExpectedResult = "union (_data | where todouble(['dayOfWeek'].['A']) >= 0 and todouble(['dayOfWeek'].['A']) < 800 | summarize wibble | extend ['key'] = '0-800'), (_data | where todouble(['dayOfWeek'].['A']) >= 800 and todouble(['dayOfWeek'].['A']) < 2000 | summarize wibble | extend ['key'] = '800-2000') | project-reorder ['key'], * asc")]
+        [TestCase(null, 800, 800, 2000, ExpectedResult = "union (_data | where todouble(['dayOfWeek'].['A']) < 800 | summarize wibble | extend ['key'] = '-800'), (_data | where todouble(['dayOfWeek'].['A']) >= 800 and todouble(['dayOfWeek'].['A']) < 2000 | summarize wibble | extend ['key'] = '800-2000') | project-reorder ['key'], * asc")]
+        [TestCase(0, 800, 800, null, ExpectedResult = "union (_data | where todouble(['dayOfWeek'].['A']) >= 0 and todouble(['dayOfWeek'].['A']) < 800 | summarize wibble | extend ['key'] = '0-800'), (_data | where todouble(['dayOfWeek'].['A']) >= 800 | summarize wibble | extend ['key'] = '800-') | project-reorder ['key'], * asc")]
+        [TestCase(null, 800, 800, null, ExpectedResult = "union (_data | where todouble(['dayOfWeek'].['A']) < 800 | summarize wibble | extend ['key'] = '-800'), (_data | where todouble(['dayOfWeek'].['A']) >= 800 | summarize wibble | extend ['key'] = '800-') | project-reorder ['key'], * asc")]
+        [TestCase(0, 10000, 2000, 3000, ExpectedResult = "union (_data | where todouble(['dayOfWeek'].['A']) >= 0 and todouble(['dayOfWeek'].['A']) < 10000 | summarize wibble | extend ['key'] = '0-10000'), (_data | where todouble(['dayOfWeek'].['A']) >= 2000 and todouble(['dayOfWeek'].['A']) < 3000 | summarize wibble | extend ['key'] = '2000-3000') | project-reorder ['key'], * asc")]
+        public string RangeVisit_WithAggregation_Dynamic_ReturnsValidResponse(double? from1, double? to1, double? from2, double? to2)
+        {
+            var rangeAggregation = new RangeAggregation() {
+                Metric = "wibble",
+                Field = "dayOfWeek.A",
+                Key = "key",
+                Ranges = new List<RangeAggregationExpression>() {
+                    new RangeAggregationExpression { Field = "dayOfWeek.A", From = from1, To = to1 },
+                    new RangeAggregationExpression { Field = "dayOfWeek.A", From = from2, To = to2 },
+                },
+            };
+
+            var visitor = VisitorTestsUtils.CreateAndVisitRootVisitor("dayOfWeek.A", "double");
```

Comment:
l57 is redundant (twice visited)

Reply:
Here it's different, the first line visits the root visitor and the second visits the aggregator

Reply:
ok, so why do we need to visit the root?

Reply:
It initializes a bunch of stuff, including `schemaRetriever`. It's in most of the tests.